File: 49. Group Anagrams/solution.go
func groupAnagrams(strs []string) [][]string {
    l := len(strs)
    var result = make(map[string][]string)
    
    for i:=0; i<l; i++ {
        str := strings.Split(strs[i], "")
        sort.Strings(str)
        key := strings.Join(str, "")
        result[key] = append(result[key], strs[i])
    }
    
    var groups [][]string
    for _,v := range(result) {
        groups = append(groups, v)
    }
    return groups
}

File: 49. Group Anagrams/solution.py
class Solution:
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        map = {}
        for s in strs:
            s_sorted = "".join(sorted(s))
            if s_sorted in map:
                map[s_sorted].append(s)
            else:
                map[s_sorted] = [s]
        
        return [map[a] for a in map]
